File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    control = {1:"w", -1: "s", 10: "d", -10: "a"}
    answer = ""
    for i in range(1, len(numLog)):
        a = numLog[i] - numLog[i-1]
        answer += control[a]
        
    return answer

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.java
import java.util.*;

class Solution {
    public String solution(int[] numLog) {
        String result = "";
        Map<Integer, String> control = new HashMap<>();
        control.put(1, "w");
        control.put(-1, "s");
        control.put(10, "d");
        control.put(-10, "a");

        for (int i = 1; i < numLog.length; i++) {
            int diff = numLog[i] - numLog[i - 1];
            result += control.get(diff);
        }
        
        return result;
    }
}